### PR TITLE
refactored get endpoints and make protectedrouteguard

### PR DESCRIPTION
**Pull request by PedroLorenzen**
1. refactored get endpoints to use same reusable endpoint
2. made protectedrouteguard to check session before accessing each page.

### CODE REVIEW

File: shelfsense/src/components/protectedRouteGuard/ProtectedRouteGuard.jsx [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
     }
 
+
     if (!isAuthenticated) {
-        return <Navigate to="/login" replace />; // Redirect if not authenticated
+        console.error("Session expired or does not match the server session")
+        return <Navigate to="/login" replace />;
     }
-
+    console.log("Session runs with user: ", user)
```
Comment (PedroLorenzen) [inline shelfsense/src/components/protectedRouteGuard/ProtectedRouteGuard.jsx]: Let's delete this comment

File: shelfsense/src/components/components/ComponentsTable.jsx [hunk 2]
```diff
@@ -102,8 +102,8 @@ const ComponentsTable = () => {
                                 <>
                                     {components
                                         .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
-                                        .map((component) => (
```
Comment (PedroLorenzen) [inline shelfsense/src/components/components/ComponentsTable.jsx:105]: Use component instead of data again

**Review by PedroLorenzen — COMMENTED** [review]
Please add more error handling

File: shelfsense/src/util/services/getRequestService.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            console.error('Failed to fetch products');
+            return null;
+        }
+        const data = await response.json();
+        console.log(data);
+        return data;
+    } catch (error) {
+        console.error(error);
+        return null;
+    }
```
Comment (PedroLorenzen) [inline shelfsense/src/util/services/getRequestService.js]: Add better error handling

**Review by UlrikLn — APPROVED** [review]

**Review by UlrikLn — APPROVED** [review]

**Review by UlrikLn — APPROVED** [review]

**Pull request merged**